### PR TITLE
Prefer Darwin to OSX

### PR DESCRIPTION
**Pull request by sdmaclea**
Rename TARGET_OSX -> TARGET_DARWIN
Rename TargetsOSX -> TargetsDarwin

### CODE REVIEW

**Comment by CoffeeFlux** [comment]
What is the motivation for this? I suspect it would require more changes on the Mono side in files that aren't mirrored.

**Comment by BruceForstall** [comment]
Seems like [Darwin](https://en.wikipedia.org/wiki/Darwin_(operating_system)) isn't a name for the "whole" OS, just the core. Could that be more confusing?

IMO, we should just use macOS / MACOS and if Apple changes it again, we can do a big rename again. If that's too disruptive, maybe just leave OSX. After all, we still use "Windows_NT" which drives me bonkers since we haven't used that term for 20 years...

**Review by lambdageek — CHANGES_REQUESTED** [review]
Mono `TARGET_OSX` define is used to distinguish it from iOS and watchOS; renaming it to `TARGET_DARWIN` is a semantic change.

File: src/mono/configure.ac [hunk 1]
```diff
@@ -4903,9 +4903,9 @@ if test "x$target_mach" = "xyes"; then
        #error fail this for ios
        #endif
        ], [
-	   	  AC_DEFINE(TARGET_OSX,1,[The JIT/AOT targets OSX])
-          CPPFLAGS_FOR_LIBGC="$CPPFLAGS_FOR_LIBGC -DTARGET_OSX"
-          CFLAGS_FOR_LIBGC="$CFLAGS_FOR_LIBGC -DTARGET_OSX"
+	   	  AC_DEFINE(TARGET_DARWIN,1,[The JIT/AOT targets OSX])
+          CPPFLAGS_FOR_LIBGC="$CPPFLAGS_FOR_LIBGC -DTARGET_DARWIN"
+          CFLAGS_FOR_LIBGC="$CFLAGS_FOR_LIBGC -DTARGET_DARWIN"
```
Comment (lambdageek) [inline src/mono/configure.ac]: The nearby code here is distinguishing the Darwin flavor: ios/watchos/macos

**Comment by am11** [comment]
`DARWIN` (like `LINUX` and `ILLUMOS`) is the base system name. `macOS`, `iOS`, `watchOS`, `tvOS` are the OS names. `OSX` is obsolete name. IMO, we can define `DARWIN` and `MACOS` macros, and in (most) cases where we do not need to specialize macos and ios etc., `DARWIN` can be used (as we use `LINUX` and not `DEBIAN` etc.).

**Comment by sdmaclea** [comment]
> What is the motivation for this?

There was a discussion with @jkotas @am11 and @BruceForstall  where the MACOS, Darwin, OSX inconsistency was noted in coreclr.  This rename was suggested to be more immune to marketing whims.

I have no time invested in this...
I certainly can: 
+ abandon the patch...
+ revise to not touch mono (except mono/System.Private.Corlib)
+ revise to not touch mono or System.Private.CoreLib
+ revise to limit this to the coreclr/jit and rename to MACOS or Darwin.

**Comment by sdmaclea** [comment]
Abandoning patch until further feedback suggests a better course...

**Comment by BruceForstall** [comment]
FWIW, the "OSX" name is now even more obsolete than it was before, now that Apple has renumbered the new "Big Sur" OS as macOS 11 (since the "X" was a Roman numeral for version 10).

**Comment by am11** [comment]
Sure, it is either `Mac OS X`, `Mac OS 9` (with numbers), or `macOS` without. While working on https://github.com/dotnet/runtime/issues/37923, I found that `/System/Library/CoreServices/SystemVersion.plist` file and `sw_vers` utility (which reads that plist file) tell us `ProductName: Mac OS X` on Mojave and Catalina (although the name was changed to `macOS` in 2016), and Bug Sur (beta) gives us: `ProductName: macOS`. Unless there are backward compatibility concerns, we can start using `macOS` (at least in new code). #naming-is-hard

**Pull request closed without merging**